front-end/src/containers/HomePage.js

```diff
@@ -11,17 +11,34 @@ class HomePage extends Component {
     this.props.loadStories();
   }
 
+  renderSearch() {
+    for (let key in this.props.page) {
+      if (Array.isArray(this.props.page[key])) delete this.props.page[key]
+    }
+    console.log(this.props.page);
+    if (Object.keys(this.props.page).length === 0) {
+      console.log('Default results');
+      return <StoryList stories={this.props.stories}/>
+    } else {
+      console.log('Search results');
+      const stories = this.props.page.searchResult.entities.stories
```

Comment:
entities stories? nope, this should be an array. When getting the results of the search request, you should work with the `results` part, not the `entities`.